### PR TITLE
chore: cleanup usage of double equals

### PR DESCRIPTION
**Pull request by lino-levan**
Originally reported on discord. A lot of our vendored implementations use double equals which is unfortunate.

### CODE REVIEW

**Comment by iuioiua** [comment]
It might be also worth getting rid of instances of `!=`.

**Review by iuioiua — COMMENTED** [review]
Does this cover all instances of `==` and `!=`? I see some cases other within JSDocs, which could also be corrected for clarity.

**Comment by lino-levan** [comment]
I wasn't sure if it was valuable to update the JSDocs since many of them just assume that `==` and `===` are the same (or `!=` and `!==`). We could update them if you think it's valuable to do so.

**Comment by iuioiua** [comment]
Updating those would provide extra context for those who understand that level of "type-strictness ".

File: log/handlers.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export class BaseHandler {
       const value = logRecord[p1 as keyof LogRecord];
 
       // do not interpolate missing values
-      if (value == null) {
+      if (value === null) {
```
Comment (kt3k) [inline log/handlers.ts:45]: This looks like a check for missing property of the object. I think this should compare with `undefined`
```suggestion
      if (value === undefined) {
```

File: media_types/_util.ts [hunk 2]
```diff
@@ -109,7 +109,7 @@ function indexOf<T>(s: Iterable<T>, fn: (s: T) => boolean): number {
 }
 
 export function isIterator<T>(obj: unknown): obj is Iterable<T> {
-  if (obj == null) {
+  if (obj === null) {
```
Comment (kt3k) [inline media_types/_util.ts:112]: Maybe this should be `obj === null || obj === undefined`?

File: path/glob_test.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ function match(
   path: string,
   opts: GlobToRegExpOptions = {},
 ): boolean {
-  if (opts.os == null) {
+  if (opts.os === null) {
```
Comment (kt3k) [inline path/glob_test.ts:17]:
```suggestion
  if (opts.os === undefined) {
```

File: semver/is_semver.ts [hunk 4]
```diff
@@ -19,7 +19,7 @@ import { isValidNumber, isValidString } from "./_shared.ts";
  * @returns True if value is a valid SemVer otherwise false
  */
 export function isSemVer(value: unknown): value is SemVer {
-  if (value == null) return false;
+  if (value === null) return false;
```
Comment (kt3k) [inline semver/is_semver.ts:22]:
```suggestion
  if (value === null || value === undefined) return false;
```

File: semver/is_semver_comparator.ts [hunk 5]
```diff
@@ -15,7 +15,7 @@ import { ALL, NONE } from "./constants.ts";
  * @returns True if the object is a SemVerComparator otherwise false
  */
 export function isSemVerComparator(value: unknown): value is SemVerComparator {
-  if (value == null) return false;
+  if (value === null) return false;
```
Comment (kt3k) [inline semver/is_semver_comparator.ts:18]:
```suggestion
  if (value === null || value === undefined) return false;
```

File: semver/is_semver_range.ts [hunk 6]
```diff
@@ -13,7 +13,7 @@ import { isSemVerComparator } from "./is_semver_comparator.ts";
  * @returns True if its a valid SemVerRange otherwise false.
  */
 export function isSemVerRange(value: unknown): value is SemVerRange {
-  if (value == null) return false;
+  if (value === null) return false;
```
Comment (kt3k) [inline semver/is_semver_range.ts:16]:
```suggestion
  if (value === null || value === undefined) return false;
```

File: fs/walk.ts
Comment (iuioiua) [inline fs/walk.ts:1]: I gather these lines were removed simply because they were redundant. Is that right?

**Review by iuioiua — APPROVED** [review]
Nice!

File: fs/walk.ts
Comment (lino-levan) [reply fs/walk.ts:1]: That's correct, yes.

File: media_types/_util.ts (hunk 2 above)
Comment (kt3k) [reply media_types/_util.ts:112]:
```suggestion
  if (obj === null || obj === undefined) {
```

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**